crates/volta-core/src/shim.rs

```diff
@@ -89,6 +89,23 @@ pub fn create(shim_name: &str) -> Fallible<ShimResult> {
     }
 }
 
+pub fn check_reachable(shim_name: &str) -> Fallible<()> {
+    let shim = volta_home()?.shim_file(shim_name);
+    let resolved = which::which(shim_name).map_err(|e| {
+        VoltaError::from_source(
+            e,
+            ErrorKind::ShimResolveError {
+                name: shim_name.to_string(),
+            },
+        )
+    })?;
+    if resolved != shim {
+        warn!("{} is shadowed by another binary of the same name at {}. Please ensure that {} is at the front of your PATH.", shim_name, resolved.display(), shim.display());
```

Comment:
Given where this is used, rather than `warn!`, I would lean a little towards using `info!` and `note_prefix`, as that's the approach we've taken for a similar message when you install a global version of Node but the local version is different. In both cases, the message exists to let you know that while you just installed something, what you get when you run the command may not be what you expect.